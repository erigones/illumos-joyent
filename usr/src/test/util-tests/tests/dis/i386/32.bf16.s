/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2024 Oxide Computer Company
 */

/*
 * Binary floating point 16 instructions, currently just AVX related (i.e. no AMX).
 *
 * For the vcvtneps2bf16 instruction, gas sometimes has a variant with 'x' or
 * 'y' appended which appear to be an indication for the target memory size,
 * particularly for broadcasts and related. While we use those (as there's no
 * other way to get that), dis currently does not break these apart.
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	vcvtne2ps2bf16	%xmm0, %xmm1, %xmm2
	vcvtne2ps2bf16	%xmm3, %xmm4, %xmm5{%k1}
	vcvtne2ps2bf16	%xmm6, %xmm7, %xmm0{%k2}{z}
	vcvtne2ps2bf16	(%eax), %xmm1, %xmm2
	vcvtne2ps2bf16	0x77(%ebx), %xmm1, %xmm2
	vcvtne2ps2bf16	0x77(%ecx,%edx,4), %xmm1, %xmm2
	vcvtne2ps2bf16	0x7777(%edi){1to4}, %xmm1, %xmm2
	vcvtne2ps2bf16	0x7777(%edi){1to4}, %xmm1, %xmm2{%k3}
	vcvtne2ps2bf16	0x7777(%edi){1to4}, %xmm1, %xmm2{%k3}{z}

	vcvtne2ps2bf16	%ymm0, %ymm1, %ymm2
	vcvtne2ps2bf16	%ymm3, %ymm4, %ymm5{%k1}
	vcvtne2ps2bf16	%ymm6, %ymm7, %ymm0{%k2}{z}
	vcvtne2ps2bf16	(%eax), %ymm1, %ymm2
	vcvtne2ps2bf16	0x77(%ebx), %ymm1, %ymm2
	vcvtne2ps2bf16	0x77(%ecx,%edx,4), %ymm1, %ymm2
	vcvtne2ps2bf16	0x7777(%edi){1to8}, %ymm1, %ymm2

	vcvtne2ps2bf16	%zmm0, %zmm1, %zmm2
	vcvtne2ps2bf16	%zmm3, %zmm4, %zmm5{%k1}
	vcvtne2ps2bf16	%zmm6, %zmm7, %zmm0{%k2}{z}
	vcvtne2ps2bf16	(%eax), %zmm1, %zmm2
	vcvtne2ps2bf16	0x77(%ebx), %zmm1, %zmm2
	vcvtne2ps2bf16	0x77(%ecx,%edx,4), %zmm1, %zmm2
	vcvtne2ps2bf16	0x7777(%edi){1to16}, %zmm1, %zmm2

	vcvtneps2bf16	%xmm0, %xmm1
	vcvtneps2bf16	%xmm2, %xmm3{%k4}
	vcvtneps2bf16	%xmm5, %xmm6{%k7}{z}
	vcvtneps2bf16x	(%edi), %xmm7
	vcvtneps2bf16x	0x88(%ebx), %xmm6
	vcvtneps2bf16x	0x88(%ebx,%ecx,4), %xmm5
	vcvtneps2bf16x	0x66(%ebx,%ecx,4), %xmm5{%k3}
	vcvtneps2bf16	(%esi){1to4}, %xmm6
	vcvtneps2bf16	(%esi){1to4}, %xmm6{%k6}
	vcvtneps2bf16	(%edi){1to8}, %xmm6

	vcvtneps2bf16	%ymm0, %xmm1
	vcvtneps2bf16	%ymm2, %xmm3{%k4}
	vcvtneps2bf16	%ymm5, %xmm6{%k7}{z}
	vcvtneps2bf16y	(%edi), %xmm7
	vcvtneps2bf16y	0x88(%ebx), %xmm6
	vcvtneps2bf16y	0x88(%ebx,%ecx,4), %xmm5
	vcvtneps2bf16y	-0x66(%ebx,%ecx,4), %xmm5{%k3}
	vcvtneps2bf16	(%esi){1to8}, %xmm6
	vcvtneps2bf16	(%esi){1to8}, %xmm6{%k6}

	vcvtneps2bf16	%zmm0, %ymm1
	vcvtneps2bf16	%zmm2, %ymm3{%k4}
	vcvtneps2bf16	%zmm5, %ymm6{%k7}{z}
	vcvtneps2bf16	(%edi), %ymm7
	vcvtneps2bf16	0x88(%ebx), %ymm6
	vcvtneps2bf16	0x88(%ebx,%ecx,4), %ymm5
	vcvtneps2bf16	-0x66(%ebx,%ecx,4), %ymm5{%k3}
	vcvtneps2bf16	(%esi){1to16}, %ymm6
	vcvtneps2bf16	(%esi){1to16}, %ymm6{%k6}

	vdpbf16ps	%xmm0, %xmm1, %xmm2
	vdpbf16ps	%xmm3, %xmm4, %xmm5{%k1}
	vdpbf16ps	%xmm6, %xmm7, %xmm0{%k2}{z}
	vdpbf16ps	(%eax), %xmm1, %xmm2
	vdpbf16ps	0x77(%ebx), %xmm1, %xmm2
	vdpbf16ps	0x77(%ecx,%edx,4), %xmm1, %xmm2
	vdpbf16ps	0x7777(%edi){1to4}, %xmm1, %xmm2
	vdpbf16ps	0x7777(%edi){1to4}, %xmm1, %xmm2{%k3}
	vdpbf16ps	0x7777(%edi){1to4}, %xmm1, %xmm2{%k3}{z}

	vdpbf16ps	%ymm0, %ymm1, %ymm2
	vdpbf16ps	%ymm3, %ymm4, %ymm5{%k1}
	vdpbf16ps	%ymm6, %ymm7, %ymm0{%k2}{z}
	vdpbf16ps	(%eax), %ymm1, %ymm2
	vdpbf16ps	0x77(%ebx), %ymm1, %ymm2
	vdpbf16ps	0x77(%ecx,%edx,4), %ymm1, %ymm2
	vdpbf16ps	0x7777(%edi){1to8}, %ymm1, %ymm2

	vdpbf16ps	%zmm0, %zmm1, %zmm2
	vdpbf16ps	%zmm3, %zmm4, %zmm5{%k1}
	vdpbf16ps	%zmm6, %zmm7, %zmm0{%k2}{z}
	vdpbf16ps	(%eax), %zmm1, %zmm2
	vdpbf16ps	0x77(%ebx), %zmm1, %zmm2
	vdpbf16ps	0x77(%ecx,%edx,4), %zmm1, %zmm2
	vdpbf16ps	0x7777(%edi){1to16}, %zmm1, %zmm2
.size libdis_test, [.-libdis_test]
