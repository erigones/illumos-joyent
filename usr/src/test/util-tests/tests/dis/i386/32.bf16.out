    libdis_test:       62 f2 77 08 72 d0  vcvtneps2bf16 %xmm0,%xmm1,%xmm2
    libdis_test+0x6:   62 f2 5f 09 72 eb  vcvtneps2bf16 %xmm3,%xmm4,%xmm5{%k1}
    libdis_test+0xc:   62 f2 47 8a 72 c6  vcvtneps2bf16 %xmm6,%xmm7,%xmm0{%k2}{z}
    libdis_test+0x12:  62 f2 77 08 72 10  vcvtneps2bf16 (%eax),%xmm1,%xmm2
    libdis_test+0x18:  62 f2 77 08 72 93  vcvtneps2bf16 0x77(%ebx),%xmm1,%xmm2
                       77 00 00 00 
    libdis_test+0x22:  62 f2 77 08 72 94  vcvtneps2bf16 0x77(%ecx,%edx,4),%xmm1,%xmm2
                       91 77 00 00 00 
    libdis_test+0x2d:  62 f2 77 18 72 97  vcvtneps2bf16 0x7777(%edi){1to4},%xmm1,%xmm2
                       77 77 00 00 
    libdis_test+0x37:  62 f2 77 1b 72 97  vcvtneps2bf16 0x7777(%edi){1to4},%xmm1,%xmm2{%k3}
                       77 77 00 00 
    libdis_test+0x41:  62 f2 77 9b 72 97  vcvtneps2bf16 0x7777(%edi){1to4},%xmm1,%xmm2{%k3}{z}
                       77 77 00 00 
    libdis_test+0x4b:  62 f2 77 28 72 d0  vcvtneps2bf16 %ymm0,%ymm1,%ymm2
    libdis_test+0x51:  62 f2 5f 29 72 eb  vcvtneps2bf16 %ymm3,%ymm4,%ymm5{%k1}
    libdis_test+0x57:  62 f2 47 aa 72 c6  vcvtneps2bf16 %ymm6,%ymm7,%ymm0{%k2}{z}
    libdis_test+0x5d:  62 f2 77 28 72 10  vcvtneps2bf16 (%eax),%ymm1,%ymm2
    libdis_test+0x63:  62 f2 77 28 72 93  vcvtneps2bf16 0x77(%ebx),%ymm1,%ymm2
                       77 00 00 00 
    libdis_test+0x6d:  62 f2 77 28 72 94  vcvtneps2bf16 0x77(%ecx,%edx,4),%ymm1,%ymm2
                       91 77 00 00 00 
    libdis_test+0x78:  62 f2 77 38 72 97  vcvtneps2bf16 0x7777(%edi){1to8},%ymm1,%ymm2
                       77 77 00 00 
    libdis_test+0x82:  62 f2 77 48 72 d0  vcvtneps2bf16 %zmm0,%zmm1,%zmm2
    libdis_test+0x88:  62 f2 5f 49 72 eb  vcvtneps2bf16 %zmm3,%zmm4,%zmm5{%k1}
    libdis_test+0x8e:  62 f2 47 ca 72 c6  vcvtneps2bf16 %zmm6,%zmm7,%zmm0{%k2}{z}
    libdis_test+0x94:  62 f2 77 48 72 10  vcvtneps2bf16 (%eax),%zmm1,%zmm2
    libdis_test+0x9a:  62 f2 77 48 72 93  vcvtneps2bf16 0x77(%ebx),%zmm1,%zmm2
                       77 00 00 00 
    libdis_test+0xa4:  62 f2 77 48 72 94  vcvtneps2bf16 0x77(%ecx,%edx,4),%zmm1,%zmm2
                       91 77 00 00 00 
    libdis_test+0xaf:  62 f2 77 58 72 97  vcvtneps2bf16 0x7777(%edi){1to16},%zmm1,%zmm2
                       77 77 00 00 
    libdis_test+0xb9:  62 f2 7e 08 72 c8  vcvtneps2bf16 %xmm0,%xmm1
    libdis_test+0xbf:  62 f2 7e 0c 72 da  vcvtneps2bf16 %xmm2,%xmm3{%k4}
    libdis_test+0xc5:  62 f2 7e 8f 72 f5  vcvtneps2bf16 %xmm5,%xmm6{%k7}{z}
    libdis_test+0xcb:  62 f2 7e 08 72 3f  vcvtneps2bf16 (%edi),%xmm7
    libdis_test+0xd1:  62 f2 7e 08 72 b3  vcvtneps2bf16 0x88(%ebx),%xmm6
                       88 00 00 00 
    libdis_test+0xdb:  62 f2 7e 08 72 ac  vcvtneps2bf16 0x88(%ebx,%ecx,4),%xmm5
                       8b 88 00 00 00 
    libdis_test+0xe6:  62 f2 7e 0b 72 ac  vcvtneps2bf16 0x66(%ebx,%ecx,4),%xmm5{%k3}
                       8b 66 00 00 00 
    libdis_test+0xf1:  62 f2 7e 18 72 36  vcvtneps2bf16 (%esi){1to4},%xmm6
    libdis_test+0xf7:  62 f2 7e 1e 72 36  vcvtneps2bf16 (%esi){1to4},%xmm6{%k6}
    libdis_test+0xfd:  62 f2 7e 38 72 37  vcvtneps2bf16 (%edi){1to8},%ymm6
    libdis_test+0x103: 62 f2 7e 28 72 c8  vcvtneps2bf16 %ymm0,%ymm1
    libdis_test+0x109: 62 f2 7e 2c 72 da  vcvtneps2bf16 %ymm2,%ymm3{%k4}
    libdis_test+0x10f: 62 f2 7e af 72 f5  vcvtneps2bf16 %ymm5,%ymm6{%k7}{z}
    libdis_test+0x115: 62 f2 7e 28 72 3f  vcvtneps2bf16 (%edi),%ymm7
    libdis_test+0x11b: 62 f2 7e 28 72 b3  vcvtneps2bf16 0x88(%ebx),%ymm6
                       88 00 00 00 
    libdis_test+0x125: 62 f2 7e 28 72 ac  vcvtneps2bf16 0x88(%ebx,%ecx,4),%ymm5
                       8b 88 00 00 00 
    libdis_test+0x130: 62 f2 7e 2b 72 ac  vcvtneps2bf16 -0x66(%ebx,%ecx,4),%ymm5{%k3}
                       8b 9a ff ff ff 
    libdis_test+0x13b: 62 f2 7e 38 72 36  vcvtneps2bf16 (%esi){1to8},%ymm6
    libdis_test+0x141: 62 f2 7e 3e 72 36  vcvtneps2bf16 (%esi){1to8},%ymm6{%k6}
    libdis_test+0x147: 62 f2 7e 48 72 c8  vcvtneps2bf16 %zmm0,%zmm1
    libdis_test+0x14d: 62 f2 7e 4c 72 da  vcvtneps2bf16 %zmm2,%zmm3{%k4}
    libdis_test+0x153: 62 f2 7e cf 72 f5  vcvtneps2bf16 %zmm5,%zmm6{%k7}{z}
    libdis_test+0x159: 62 f2 7e 48 72 3f  vcvtneps2bf16 (%edi),%zmm7
    libdis_test+0x15f: 62 f2 7e 48 72 b3  vcvtneps2bf16 0x88(%ebx),%zmm6
                       88 00 00 00 
    libdis_test+0x169: 62 f2 7e 48 72 ac  vcvtneps2bf16 0x88(%ebx,%ecx,4),%zmm5
                       8b 88 00 00 00 
    libdis_test+0x174: 62 f2 7e 4b 72 ac  vcvtneps2bf16 -0x66(%ebx,%ecx,4),%zmm5{%k3}
                       8b 9a ff ff ff 
    libdis_test+0x17f: 62 f2 7e 58 72 36  vcvtneps2bf16 (%esi){1to16},%zmm6
    libdis_test+0x185: 62 f2 7e 5e 72 36  vcvtneps2bf16 (%esi){1to16},%zmm6{%k6}
    libdis_test+0x18b: 62 f2 76 08 52 d0  vdpbf16ps %xmm0,%xmm1,%xmm2
    libdis_test+0x191: 62 f2 5e 09 52 eb  vdpbf16ps %xmm3,%xmm4,%xmm5{%k1}
    libdis_test+0x197: 62 f2 46 8a 52 c6  vdpbf16ps %xmm6,%xmm7,%xmm0{%k2}{z}
    libdis_test+0x19d: 62 f2 76 08 52 10  vdpbf16ps (%eax),%xmm1,%xmm2
    libdis_test+0x1a3: 62 f2 76 08 52 93  vdpbf16ps 0x77(%ebx),%xmm1,%xmm2
                       77 00 00 00 
    libdis_test+0x1ad: 62 f2 76 08 52 94  vdpbf16ps 0x77(%ecx,%edx,4),%xmm1,%xmm2
                       91 77 00 00 00 
    libdis_test+0x1b8: 62 f2 76 18 52 97  vdpbf16ps 0x7777(%edi){1to4},%xmm1,%xmm2
                       77 77 00 00 
    libdis_test+0x1c2: 62 f2 76 1b 52 97  vdpbf16ps 0x7777(%edi){1to4},%xmm1,%xmm2{%k3}
                       77 77 00 00 
    libdis_test+0x1cc: 62 f2 76 9b 52 97  vdpbf16ps 0x7777(%edi){1to4},%xmm1,%xmm2{%k3}{z}
                       77 77 00 00 
    libdis_test+0x1d6: 62 f2 76 28 52 d0  vdpbf16ps %ymm0,%ymm1,%ymm2
    libdis_test+0x1dc: 62 f2 5e 29 52 eb  vdpbf16ps %ymm3,%ymm4,%ymm5{%k1}
    libdis_test+0x1e2: 62 f2 46 aa 52 c6  vdpbf16ps %ymm6,%ymm7,%ymm0{%k2}{z}
    libdis_test+0x1e8: 62 f2 76 28 52 10  vdpbf16ps (%eax),%ymm1,%ymm2
    libdis_test+0x1ee: 62 f2 76 28 52 93  vdpbf16ps 0x77(%ebx),%ymm1,%ymm2
                       77 00 00 00 
    libdis_test+0x1f8: 62 f2 76 28 52 94  vdpbf16ps 0x77(%ecx,%edx,4),%ymm1,%ymm2
                       91 77 00 00 00 
    libdis_test+0x203: 62 f2 76 38 52 97  vdpbf16ps 0x7777(%edi){1to8},%ymm1,%ymm2
                       77 77 00 00 
    libdis_test+0x20d: 62 f2 76 48 52 d0  vdpbf16ps %zmm0,%zmm1,%zmm2
    libdis_test+0x213: 62 f2 5e 49 52 eb  vdpbf16ps %zmm3,%zmm4,%zmm5{%k1}
    libdis_test+0x219: 62 f2 46 ca 52 c6  vdpbf16ps %zmm6,%zmm7,%zmm0{%k2}{z}
    libdis_test+0x21f: 62 f2 76 48 52 10  vdpbf16ps (%eax),%zmm1,%zmm2
    libdis_test+0x225: 62 f2 76 48 52 93  vdpbf16ps 0x77(%ebx),%zmm1,%zmm2
                       77 00 00 00 
    libdis_test+0x22f: 62 f2 76 48 52 94  vdpbf16ps 0x77(%ecx,%edx,4),%zmm1,%zmm2
                       91 77 00 00 00 
    libdis_test+0x23a: 62 f2 76 58 52 97  vdpbf16ps 0x7777(%edi){1to16},%zmm1,%zmm2
                       77 77 00 00 
